bitcoin/script/script.h

```diff
@@ -587,4 +573,28 @@ struct CScriptWitness
 
 bool CheckMinimalPush(const std::vector<unsigned char>& data, opcodetype opcode);
 
+/** Build a script by concatenating other scripts, or any argument accepted by CScript::operator<<. */
+template<typename... Ts>
+CScript BuildScript(Ts&&... inputs)
+{
+    CScript ret;
+    size_t cnt{0}; // Try to optimize over repeatedly calling ret.empty()
+
+    ([&] (Ts&& input) {
+        if constexpr (std::is_same_v<std::remove_cv_t<std::remove_reference_t<Ts>>, CScript>) {
```

Comment:
I think this would also work, and is perhaps more readable `if constexpr (std::is_same_v<const Ts&, const Cscript&>) {`.

Reply:
This fails to compile.
```cpp
./script/script.h: In instantiation of ‘CScript BuildScript(Ts&& ...) [with Ts = {opcodetype, CScript&, opcodetype}]’:                                                                                                                                                                                                                                                     
./script/miniscript.h:525:58:   required from ‘CScript miniscript::Node<Key>::ToScript(const Ctx&) const [with Ctx = {anonymous}::KeyConverter; Key = CPubKey]’                                                                                                                                                                                                            
test/miniscript_tests.cpp:317:33:   required from here                                                                                                                                                                                                                                                                                                                     
./script/script.h:597:17: error: use of deleted function ‘CScript& CScript::operator<<(const CScript&)’                                                                                                                                                                                                                                                                    
  597 |             ret << input;                                                                                                                                                                                                                                                                                                                                          
      |             ~~~~^~~~~~~~                                                                                                                                                                                                                                                                                                                                           
./script/script.h:442:14: note: declared here                                                                                                                                                                                                                                                                                                                              
  442 |     CScript& operator<<(const CScript& b) = delete;
```

Reply:
Strange, but I'm not going to bother figuring out why now.